servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLServer.kt

```diff
@@ -39,18 +45,21 @@ open class GraphQLServer<Request>(
      * In the case of errors or exceptions, return a response with GraphQLErrors populated.
      * If you need custom logic inside this method you can override this class or choose not to use it.
      */
-    open suspend fun execute(request: Request): GraphQLServerResponse? {
+    open suspend fun execute(request: Request): GraphQLServerResponse? = coroutineScope {
         val graphQLRequest = requestParser.parseRequest(request)
 
-        return if (graphQLRequest != null) {
-            val context = contextFactory.generateContext(request)
-            val graphQLContext = contextFactory.generateContextMap(request)
+        if (graphQLRequest != null) {
+            val contextMap = contextFactory.generateContextMap(request) ?: emptyMap<Any, Any?>()
+
+            val customContext: CoroutineContext = contextMap[GRAPHQL_COROUTINE_CONTEXT_KEY] as? CoroutineContext ?: EmptyCoroutineContext
+            val graphQLExecutionScope = CoroutineScope(coroutineContext + customContext + SupervisorJob())
```

Comment:
Not an expert on coroutines, but does this need to be cancelled at the end of request processing?  Was just reading through this:  https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/index.html

Reply:
When we return the value from this method then all children coroutines will already complete so there is nothing to cancel.